cli/src/serve.js

```diff
@@ -267,18 +271,44 @@ const yes_no_options = [ 'debug',
                          'allow_anonymous' ];
 
 const parse_connect = (connect, config) => {
-  const host_port = connect.split(':');
-  if (host_port.length === 1) {
-    config.rdb_host = host_port[0];
-    config.rdb_port = default_rdb_port;
-  } else if (host_port.length === 2) {
-    config.rdb_host = host_port[0];
-    config.rdb_port = parseInt(host_port[1]);
-    if (isNaN(config.rdb_port) || config.rdb_port < 0 || config.rdb_port > 65535) {
-      throw new Error(`Invalid port: "${host_port[1]}".`);
+  // support rethinkdb:// style connection uri strings
+  // expects rethinkdb://host:port` at a minimum but can optionally take a user:pass and db
+  // e.g. rethinkdb://user:pass@host:port/db
+  const rdb_uri = url.parse(connect);
+  if (rdb_uri.protocol === "rethinkdb:"){
+    if (rdb_uri.hostname && rdb_uri.port) {
```

Comment:
Not exactly sure what `url.parse` does when the port is omitted, but it should be optional.  If there is no port, we should use the default rethinkdb driver port (`const default_rdb_port = 28015;`).